pkg/controller/statusmanager/pod_status.go

```diff
@@ -120,17 +128,54 @@ func (status *StatusManager) setNodeNetworkUnavailable(nodeName string, ready bo
 		Reason:             reason,
 		Message:            message,
 	}
-	SetNodeCondition(&node.Status.Conditions, condition)
-	err = status.client.Status().Update(context.TODO(), node)
-	if err != nil {
-		log.Error(err, "Failed to update node condition")
+	changed := SetNodeCondition(&node.Status.Conditions, condition)
+	if changed {
+		if ready {
+			last := time.Now()
+			_, present := sharedInfo.LastNetworkAvailabe[nodeName]
+			if !present {
+				sharedInfo.LastNetworkAvailabe[nodeName] = last
+				log.Info(fmt.Sprintf("Last network unavailable time of node %s is set to %s", nodeName, last))
+
+				go func() {
+					for {
+						start := time.Now()
+						if (sharedInfo.LastNetworkAvailabe[nodeName]) != last {
+							log.Info(fmt.Sprintf("Last network unavailable time of node %s changed, expecting %v, got %v, goroutine %v exiting", nodeName, last, sharedInfo.LastNetworkAvailabe[nodeName], start))
+							return
+						}
+						now := time.Now()
+						if int64(now.Sub(sharedInfo.LastNetworkAvailabe[nodeName])) > operatortypes.TimeBeforeRecoverNetwork * 1000000000 {
+							err = status.client.Status().Update(context.TODO(), node)
+							if err != nil {
+								log.Error(err, "Failed to update node condition to true")
+							} else {
+								log.Info("Updated node condition to true")
+							}
+							return
+						} else {
+							log.Info(fmt.Sprintf("Waiting longer to confirm the recovery of network status for node %s. Last network available time is %s", nodeName, sharedInfo.LastNetworkAvailabe[nodeName]))
+						}
+						time.Sleep(time.Second * 5)
+					}
+				}()
+			}
+
+		} else {
+			err = status.client.Status().Update(context.TODO(), node)
```

Comment:
Line 149-155 and line 165-170 are the same, can we create another func for it?

Reply:
make sense, done